lib/modules/nightMode.js

```diff
@@ -251,14 +259,34 @@ function overrideNightMode() {
 	nightmodeOverrideStorage.set(Date.now());
 }
 
-function isTimeForNightMode() {
+function getGeolocation() {
+	return new Promise(resolve => {
+		navigator.geolocation.getCurrentPosition(position => resolve(position.coords));
+	});
+}
+
+async function isTimeForNightMode() {
 	const currentDate = new Date();
 	const currentHour = currentDate.getHours();
 	const currentMinute = currentDate.getMinutes();
 	const currentTime = currentHour * 100 + currentMinute;
 
-	const startingTime = convertTimeStringToInt(module.options.nightModeStart.value);
-	const endingTime = convertTimeStringToInt(module.options.nightModeEnd.value);
+	const isAuto = module.options.automaticNightMode.value === 'automatic';
+
+	let startingTimeString, endingTimeString;
+
+	if (isAuto) {
+		const { latitude, longitude } = await getGeolocation();
```

Comment:
It would probably improve UX to optimistically guess at a time (nightModeStart/nightModeEnd by default, or last value) and then in the background fetch a new `getSunriseSunset`